src/main/kotlin/org/wycliffeassociates/otter/jvm/app/ui/collectionsgrid/viewmodel/ViewCollectionsViewModel.kt

```diff
@@ -0,0 +1,81 @@
+package org.wycliffeassociates.otter.jvm.app.ui.collectionsgrid.viewmodel
+
+import com.github.thomasnield.rxkotlinfx.observeOnFx
+import com.github.thomasnield.rxkotlinfx.toObservable
+import javafx.beans.property.SimpleBooleanProperty
+import javafx.beans.property.SimpleObjectProperty
+import javafx.beans.property.SimpleStringProperty
+import javafx.collections.FXCollections
+import javafx.collections.ObservableList
+import org.wycliffeassociates.otter.common.data.model.Collection
+import org.wycliffeassociates.otter.common.data.model.Content
+import org.wycliffeassociates.otter.common.domain.content.AccessTakes
+import org.wycliffeassociates.otter.jvm.app.ui.inject.Injector
+import tornadofx.*
+
+class ViewCollectionsViewModel: ViewModel() {
+    private val injector: Injector by inject()
+    private val directoryProvider = injector.directoryProvider
+    private val collectionRepository = injector.collectionRepo
+    private val contentRepository = injector.contentRepository
+    private val takeRepository = injector.takeRepository
+    private val pluginRepository = injector.pluginRepository
+
+    // The selected project
+    private var activeProject: Collection by property()
+    val activeProjectProperty = getProperty(ViewCollectionsViewModel::activeProject)
+
+    val projectTitle: SimpleStringProperty = SimpleStringProperty("")
+
+    // List of collection children (i.e. the chapters) to display in the list
+    var children: ObservableList<Collection> = FXCollections.observableList(mutableListOf())
+
+    // Selected child
+    private var activeCollection: Collection by property()
+    val activeCollectionProperty = getProperty(ViewCollectionsViewModel::activeCollection)
+
+    // List of content to display on the screen
+    // Boolean tracks whether the content has takes associated with it
+    val allContent: ObservableList<Pair<SimpleObjectProperty<Content>, SimpleBooleanProperty>>
+            = FXCollections.observableArrayList()
+    val filteredContent: ObservableList<Pair<SimpleObjectProperty<Content>, SimpleBooleanProperty>>
+            = FXCollections.observableArrayList()
+
+    // Whether the UI should show the plugin as active
+    private var showPluginActive: Boolean by property(false)
+    val showPluginActiveProperty = getProperty(ViewCollectionsViewModel::showPluginActive)
+
+
+    private var loading: Boolean by property(false)
+    val loadingProperty = getProperty(ViewCollectionsViewModel::loading)
+
+    // Create the use cases we need (the model layer)
+    private val accessTakes = AccessTakes(contentRepository, takeRepository)
+
+    init {
+        activeProjectProperty.toObservable().subscribe { setTitleAndChapters() }
```

Comment:
```
activeProjectProperty.toObservable().subscribe { 
    bindTitle()
    bindCollections()
}
```

Reply:
note that I use it; if possible, it would be great to not be holding some kind of state, but keep things more functional; previously the setTitleAndChapters method had no arguments, so it was assuming some kind of side effects and state

(if possible)